gyp/pylib/gyp/msvs_emulation.py

```diff
@@ -519,7 +524,7 @@ def GetLibFlags(self, config, gyp_to_build_path):
     libflags.extend(self._GetAdditionalLibraryDirectories(
         'VCLibrarianTool', config, gyp_to_build_path))
     lib('LinkTimeCodeGeneration', map={'true': '/LTCG'})
-    lib('TargetMachine', map={'1': 'X86', '17': 'X64', '3': 'ARM'},
+    lib('TargetMachine', map={'1': 'X86', '17': 'X64', '3': 'ARM', '4': 'ARM64'},
```

Comment:
It is worth noting that the enum from which these numbers are drawn, [machineTypeOption](https://docs.microsoft.com/en-us/dotnet/api/microsoft.visualstudio.vcprojectengine.machinetypeoption), does not have an official value for ARM64 yet. As I understand it this will be fixed in VS2017.

Please also note that the `link.exe` argument `/MACHINE:ARM64` *does* exist and works as expected.

Reply:
Regarding to the number I suppose that it will be increased by one, considering the current biggest 'value' in the enum, so it might be '18' unofficially. That's interesting I found this '18' value for /MACHINE:ARM64 in [cmake repoitory](https://gitlab.kitware.com/cmake/cmake/blob/master/Source/cmLocalVisualStudio7Generator.cxx#L534).
Shall I leave a FIXME comment on that as a reminder? It could be fixed, when ARM64 is added to machineTypeOption officially.

Reply:
I saw an email with a comment from you that I don't see here. `18` is a pretty good guess for now, and a simple TODO comment about ensuring it matches the machineTypeOption enum in the future would be great.

Also: I've been told the enum will be updated in VS2019, not VS2017.

Reply:
Updated the value to 18. I also left TODO comments.